doc: net: Move networking chapters around

jukkar opened the pull request:
Trying to group the networking chapters so that the layout is
more logical.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

jukkar commented:
The networking doc layout would look something like this with this PR.
![zephyr-networking-chapters](https://user-images.githubusercontent.com/1098415/53158611-b2bf6900-35cc-11e9-87f3-2e812f873de1.png)

rveerama1 commented on doc/reference/networking/apis.rst:
```diff
@@ -0,0 +1,19 @@
+   net_stats.rst
+   net_context.rst
+   promiscuous.rst
+   trickle.rst
```
Overall looks good. This section is still mixed with different things.

jukkar committed:
doc: net: Move networking reference document chapters around

Group the networking chapters for reference document so that the
layout is more logical and easier to read.

Signed-off-by: Jukka Rissanen <jukka.rissanen@linux.intel.com>

rlubos commented:
I like it, looks pretty neat. Do we want to reference Thread (OpenThread) under `Network Technologies`? I can prepare a page for that.

nashif commented:
looks good, way to go.

jukkar commented:
> I like it, looks pretty neat. Do we want to reference Thread (OpenThread) under `Network Technologies`? I can prepare a page for that.

That sounds like a good plan so please do.

I will submit a proper PR for this after the document changes that I submitted this week have been merged.

dbkinder commented:
I like too!  BTW, you can see the Shippable doc build artifacts for any PR: https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/13616/docs/index.html    (adjust the PR number in the URL)

jukkar commented:
Updated the chapter location a bit. I also noticed that there are still couple of networking reference files that are missing Overview chapter. I will send new PRs for these.

jukkar commented:
> BTW, you can see the Shippable doc build artifacts for any PR: https://builds.zephyrproject.org/zephyrproject-rtos/zephyr/13616/docs/index.html (adjust the PR number in the URL)

Cool, I did not know this. Thanks for the info!

dbkinder reviewed (approved):
thanks!

pfalcon commented on doc/reference/networking/system_mgmt.rst:
```diff
@@ -0,0 +1,18 @@
+.. _net_system_mgmt:
+
+Network System Management
```
I would add `(TSN)` at the end of title here, as it's commonly known by abbreviation.
